src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -22,11 +28,18 @@ namespace NuGet.PackageManagement.VisualStudio
     /// </summary>
     public abstract class PackageReferenceProject : BuildIntegratedNuGetProject
     {
+        internal static readonly Comparer<PackageReference> PackageReferenceMergeComparer = Comparer<PackageReference>.Create((a, b) => a?.PackageIdentity?.CompareTo(b.PackageIdentity) ?? 1);
+
+        private protected readonly Dictionary<PackageIdentity, TransitiveEntry> TransitiveOriginsCache = new();
```

Comment:
Do we need custom <PackageIdentity> comparer for this dictionary?

Reply:
[PackageIdentity](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Core/PackageIdentity.cs) implements `IEquatable<PackageIdentity>`, [so no](https://docs.microsoft.com/en-us/dotnet/api/system.iequatable-1?view=net-6.0#notes-to-implementers).